src/content/reference/react/cache.md

```diff
@@ -0,0 +1,443 @@
+---
+title: cache
+canary: true
+---
+
+<Canary>
+`cache` is currently only available in React’s [Canary](https://react.dev/community/versioning-policy#canary-channel) and [experimental](https://react.dev/community/versioning-policy#experimental-channel) channels. Please ensure you understand the limitations before using `cache` in production.
+
+Learn more about [React's release channels here](/community/versioning-policy#all-release-channels).
+
+</Canary>
+
+<Intro>
+
+`cache` lets you cache the result of a data fetch or computation.
+
+```js
+const cachedFn = cache(fn);
+```
+
+</Intro>
+
+<InlineToc />
+
+---
+
+## Reference {/*reference*/}
+
+### `cache(fn)` {/*cache*/}
+
+Wrap a function with `cache` to receive a version of that function with caching.
+
+```js {4,7}
+import {cache} from 'react';
+import fetchUser from 'lib/user';
+
+const getUser = cache(fetchUser);
+
+function UserProfile(id) {
+  const user = getUser(id);
+  //...
+}
+```
+
+`getUser` will first check if there is a cached result for `id`, otherwise call `fetchUser(id)` and cache the returned result.
+
+[See more examples below.](#usage)
+
+#### Parameters {/*parameters*/}
+
+- `fn`: The function you want to cache results for. `fn` can take any arguments and return any value.
+
+#### Returns {/*returns*/}
+
+`cache` returns a cached version of `fn` with the same type signature.
+
+When calling `cachedFn` with given arguments, it first checks if a cached result exists in the cache. If a cached result exists, it returns the result. If not, it calls `fn` with the arguments, stores the result in the cache, and returns the result.
+
+<Note>
+
+The optimization of caching return values based on inputs is known as [_memoization_](https://en.wikipedia.org/wiki/Memoization). We refer to `cachedFn` as a memoized function.
+
+</Note>
+
+#### Caveats {/*caveats*/}
+
+[//]: # 'TODO: add links to Server/Client Component reference once https://github.com/reactjs/react.dev/pull/6177 is merged'
+
+- `cache` is recommended for Server Components only. There are plans to introduce `cache` for Client Components, but it is not recommended today.
```

Comment:
> [...], but it is not recommended today.

This implies it is usable but may not work correctly. Is that the case? Or are we actively discouraging it? (And if we are, we could probably use stronger language here.)

Reply:
We are actively discouraging it. We are aiming to add a warning when using `cache` on the client. How about
```suggestion
- `cache` is recommended for Server Components only. There are plans to introduce `cache` for Client Components, but it is not a supported feature. Today, `cache` will no-op and warn if used in a Client Component.
```

That last bit I'd remove since we don't actually warn (yet) and add it in when we do. cc @acdlite